File: software/examples/go/example_blink_colon.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/segment_display_4x7_v2_bricklet"
	"time"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Segment Display 4x7 Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	sd, _ := segment_display_4x7_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	sd.SetBrightness(7) // Set to full brightness

	// Blink colon 10 times
	for i := 0; i < 10; i++ {

		// Activate segments of colon
		sd.SetSelectedSegment(32, true)
		sd.SetSelectedSegment(33, true)

		time.Sleep(250 * time.Millisecond)

		// Deactivate segments of colon
		sd.SetSelectedSegment(32, false)
		sd.SetSelectedSegment(33, false)
	}

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}

File: software/examples/vbnet/ExampleBlinkColon.vb
Imports System
Imports System.Threading
Imports Tinkerforge

Module ExampleBlinkColon
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Segment Display 4x7 Bricklet 2.0

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim sd As New BrickletSegmentDisplay4x7V2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        sd.SetBrightness(7) ' Set to full brightness

        ' Blink colon 10 times
        Dim i As Integer
        For i = 0 To 9

            ' Activate segments of colon
            sd.SetSelectedSegment(32, True)
            sd.SetSelectedSegment(33, True)

            Thread.Sleep(250)

            ' Deactivate segments of colon
            sd.SetSelectedSegment(32, False)
            sd.SetSelectedSegment(33, False)
        Next i

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module
